src/expr.rs

```diff
@@ -549,6 +550,21 @@ pub mod parsing {
         ))
     ));
 
+    named!(expr_loop -> Expr, do_parse!(
+        id: option!(
+            terminated!(
+                preceded!(
```

Comment:
You can simplify this in one of two ways:

1. Using [the existing lifetime parser](https://github.com/dtolnay/syn/blob/a3faf259f2620fc6cc38d81b96a6814716f119b6/src/generics.rs#L116-L121) like `terminated!(lifetime, punct!(":"))`.

    Then in the Expr::Loop at the bottom you will need to use `label.map(|lt| lt.ident)` to pull out the ident.

    You will need to add `use generics::parsing::lifetime` [up here](https://github.com/dtolnay/syn/blob/a3faf259f2620fc6cc38d81b96a6814716f119b6/src/expr.rs#L340-L343) to be able to use it.

2. Using the `delimited` helper which is a combination of `preceded` and `terminated`:

    `delimited!(punct("'"), ident, punct(":"))`.

    Note that in this case you need to do `format!("'{}", id).into()` below in order to address my other comment.

Up to you which one you find nicer.